app/bot.py

```diff
@@ -1,275 +1,214 @@
-				elif location['status'] == "ZERO_RESULTS":
-					pass
-					return end_conversation(root_id, max_step)
-		else:
-			pass
-
-	elif max_step.conversation_status == 5:
-		DB.update_conversation_checks(root_id)
-	elif max_step.conversation_status == 10:
-
-		processed_dms = twitter.process_dms(user_id=max_step.in_reply_to_id, tweet_id=max_step.tweet_id, incident_id=max_step.incident_id, convo_tree_txt=conversation_tree[11])
-		if processed_dms is not None:
-
-			to_insert = {}
-			to_insert["incident_id"] = max_step.incident_id
-			to_insert["form"] = max_step.form
-			to_insert['tweeter_id'] = processed_dms['tweeter_id']
-			to_insert['tweet_text'] = processed_dms['quick_reply_response']
-			to_insert['tweet_id'] = max_step.tweet_id
-			to_insert['reachout_template'] = conversation_tree[11]
-			to_insert['checks_made'] = (max_step.checks_made+1)
-			to_insert['conversation_status'] = processed_dms['conversation_status']
-
-			DB.insert_data_conversations([to_insert])
-
-	elif max_step.conversation_status == 12:
-		# This is where we've received a response, shouldn't do anything here, only return when asked for directly through endpoint
-		DB.update_conversation_checks(root_id)
-
-	elif max_step.conversation_status == 13:
-		DB.update_conversation_checks(root_id)
+    """ Advances conversation by root_id """
+    api = create_api()
+    root_conversation = DB.get_conversation_root(root_id)
+    # TODO Consider refactoring so it is not a list. 
+    # (Depends on how many users can contribute to incident)
+    for steps in root_conversation:  # This checks every item in Conversations Table by tweet id
+        max_step = steps
+
+    if max_step.conversation_status == 0 and max_step.form == 0:
+        # Currently send_form handles this function
+        # This is really for conversation processing
+        status = twitter.respond_to_tweet(max_step.tweet_id, conversation_tree[1])
+        max_step.tweet_id = root_id
+        max_step.sent_tweet_id = status.id_str
+        max_step.in_reply_to_id = status.in_reply_to_status_id
+        max_step.conversation_status = (max_step.conversation_status + 1)
+        max_step.checks_made = (max_step.checks_made + 1)
+        max_step.reachout_template = conversation_tree[1]
+        DB.insert_data_conversations(max_step)
+
+    elif max_step.conversation_status == 0 and max_step.form == 1:
+        status = twitter.respond_to_tweet(max_step.tweet_id, max_step.form_link)
+        max_step.tweet_id = root_id
+        max_step.sent_tweet_id = status.id_str
+        max_step.in_reply_to_id = status.in_reply_to_status_id
+        max_step.conversation_status = 4
+        max_step.checks_made = (max_step.checks_made + 1)
+        max_step.reachout_template = form_link
+        DB.insert_data_conversations(max_step)
+
+    # Possible TODO Classification or NLP Model can be implemented here for tweet responses
+    elif max_step.conversation_status == 1 and max_step.form == 0:
+
```

Comment:
I know alot of these changes have come from previous work, we need to get these added for sure.